test/python/utest_pymasar/test_addmasarconfigs.py

```diff
@@ -0,0 +1,136 @@
+            self.parsed_json = json.load(file)
+
+    def tearDown(self):
+        pass
+
+    def testSaveService(self):
+        if self.db == "sqlite":
+            # DB SETUP
+            __sqlitedb__ = os.environ["MASAR_SQLITE_DB"]
+            try:
+                conn = sqlite3.connect(__sqlitedb__)
+                cur = conn.cursor()
+                __sql__ = None
+                if __sql__ is None:
+                    from pymasarsqlite.db.masarsqlite import SQL
+                else:
+                    sqlfile = open(__sql__)
+                    SQL = sqlfile.read()
+                if SQL is None:
+                    print ('SQLite script is empty. Cannot create SQLite db.')
+                    sys.exit()
+                else:
+                    cur.executescript(SQL)
+                    cur.execute("PRAGMA main.page_size= 4096;")
+                    cur.execute("PRAGMA main.default_cache_size= 10000;")
+                    cur.execute("PRAGMA main.locking_mode=EXCLUSIVE;")
+                    cur.execute("PRAGMA main.synchronous=NORMAL;")
+                    cur.execute("PRAGMA main.journal_mode=WAL;")
+                    cur.execute("PRAGMA main.temp_store = MEMORY;")
+
+                cur.execute('select name from sqlite_master where type=\'table\'')
+                masarconf = 'SR_All_20140421'
+                servicename = 'masar'
+
+                pvgname = 'masarpvgroup'
+                pvgdesc = 'this is my new pv group for masar service with same group name'
+                pvs = ["masarExample0000",
+                       "masarExample0001",
+                       "masarExampleBoUninit",
+                       "masarExampleMbboUninit",
+                       "masarExample0002",
+                       "masarExample0003",
+                       "masarExample0004",
+                       "masarExampleCharArray",
+                       "masarExampleShortArray",
+                       "masarExampleLongArray",
+                       "masarExampleStringArray",
+                       "masarExampleFloatArray",
+                       "masarExampleDoubleArray",
+                       "masarExampleMbboUninitTest"]
+                res = savePvGroup(conn, pvgname, func=pvgdesc)
+                res = saveGroupPvs(conn, pvgname, pvs)
+                pvgroups = retrievePvGroups(conn)
+                saveService(conn, servicename, desc='test desc')
+                conn.commit()
+                conn.close()
+                savePvGroups(self.parsed_json['pvgroups'])
+                saveSQLiteServiceConfig(self.parsed_json)
+                conn = sqlite3.connect(__sqlitedb__)
+                pvgroups = retrievePvGroups(conn)
+                self.assertEqual('BR_MG_Set_20130419', pvgroups[1][1])
+                self.assertEqual('Booster magnet power supply set points', pvgroups[1][2])
+                self.assertEqual(3, len(pvgroups[1][3].split(':')))  # Confirms correct date format
```

Comment:
To enhance maintainability, when "parsing" like this it is a friendly to put a comment with an example of the string should look like.  This will help in future if the test starts failing to determine if the input string format has changed, or if the parser is wrong/incomplete.